### PR TITLE
test-configs.yaml: Add more LTP components with appropriate timeout v…

### PR DESCRIPTION
**Pull request by Lakshmipathi**
…alue

Signed-off-by: Lakshmipathi <lakshmipathi.ganapathi@collabora.com>

### CODE REVIEW

File: test-configs.yaml [hunk 1]
```diff
@@ -1566,7 +1940,41 @@ test_configs:
   - device_type: bcm2836-rpi-2-b
     test_plans:
       - baseline
+      - ltp-admin_tools
```
Comment (gctucker) [inline test-configs.yaml:1943]: For test configs, we can easily have a YAML anchor as we can have multiple test config entries with the same device type.  For example:
```yaml
ltp-test-plans: &ltp_test_plans
  - ltp-admin_tools
  - ltp-connectors
  - ltp-containers
# etc...

test_configs:
  - device_type: bcm2836-rpi-2-b
    test_plans:
      - baseline
      - sleep
      - usb

  - device_type: bcm2836-rpi-2-b
    test_plans: *ltp_test_plans
```

Comment (gctucker) [reply test-configs.yaml:1943]: Also we'll see how things go when running everything on all the test devices, maybe we'll want to also have anchors with some LTP subsets.  For example, `ltp-fs-plans`, `ltp-mem-plans` etc... with just a handful of things to run in order to optimise how labs' capacity is being used.  But that's for a bit later, once we have started collecting results.  It's probably going to be an on-going process in fact, with new labs and new devices and new versions of LTP.

**Review by gctucker — COMMENTED** [review]
Looks like a good starting point to start running LTP more widely and get more results on staging.  We should however at least use YAML anchors to make this easier to maintain, and later on improve how the configuration is managed while this keeps growing.

**Comment by gctucker** [comment]
This PR conflicts with earlier PRs and as such it's not currently being deployed on staging.  That's why the LTP results don't appear on the web frontend yet and no jobs are being submitted.  We should try to get this run at least once this week.

File: test-configs.yaml [hunk 2]
```diff
@@ -170,6 +170,193 @@ test_plans:
     params:
       job_timeout: '120'
 
+  ltp-admin_tools:
```
[the remark this answers is not in the document]
Comment (Lakshmipathi) [reply test-configs.yaml]: okay added this change and verified config files.

File: test-configs.yaml (hunk 1 above)
Comment (Lakshmipathi) [reply test-configs.yaml]: > For test configs, we can easily have a YAML anchor as we can have multiple test config entries with the same device type. For example:
> 
> ```yaml
> ltp-test-plans: &ltp_test_plans
>   - ltp-admin_tools
>   - ltp-connectors
>   - ltp-containers
> # etc...
> 
> test_configs:
>   - device_type: bcm2836-rpi-2-b
>     test_plans:
>       - baseline
>       - sleep
>       - usb
> 
>   - device_type: bcm2836-rpi-2-b
>     test_plans: *ltp_test_plans
> ```

updated the patch with this change.

File: test-configs.yaml [hunk 3]
```diff
@@ -1468,6 +1721,7 @@ device_types:
       - blocklist: {defconfig: ['multi_v7_defconfig+linaro-base+distribution']}
 
 
+
```
Comment (gctucker) [inline test-configs.yaml:1724]: Superfluous blank line...

File: test-configs.yaml [hunk 4]
```diff
@@ -224,6 +440,43 @@ device_default_filters:
...
+      - ltp-pty
+      - ltp-sched
+      - ltp-security
+      - ltp-sound
+      - ltp-syscalls
+      - ltp-timers
+      - ltp-tracing
+      - ltp-uevents
+
 device_types:
```
Comment (gctucker) [inline test-configs.yaml:480]: This is where having 2 blank lines helps :)

File: test-configs.yaml [hunk 5]
```diff
@@ -170,17 +170,233 @@ test_plans:
...
+  ltp-opts: &ltp-rootfs
     rootfs: debian_buster-ltp_nfs
     pattern: 'ltp/{category}-{method}-{protocol}-{rootfs}-ltp-template.jinja2'
     filters:
       - blocklist: *kselftest_defconfig_filter
-    params:
+    params: &ltp-params
       skip_install: "true"
-      tst_cmdfiles: "ipc"
+      tst_cmdfiles: "ltp"
```
Comment (gctucker) [inline test-configs.yaml:180]: Maybe use an empty string here with a comment rather than `ltp` as it's not a real subset name.

**Review by gctucker — COMMENTED** [review]
Thanks, that's much better as there's a lot less duplication.  We'll probably need to split this config file into separate ones at some point, but for now it's fine.

Next step is to get it to run on staging, I think we have to merge the `ltp-crypto` PR #538 first as this one seems to be conflicting with it.

**Comment by gctucker** [comment]
Also the summary in the git commit seems to be a bit too long.  Maybe reformat it like this:
```
test-configs.yaml: define more LTP test plans

[describe in more detail what this commit does]

Signed-off-by: ...
```

**Comment by gctucker** [comment]
@Lakshmipathi Maybe this PR should be closed if you're now going to be adding each test plan one at a time?

**Comment by Lakshmipathi** [comment]
Yes @gctucker will be adding each test plan one at a time. Closing this now.

**Pull request closed without merging**